Corona-Warn-App/src/main/java/de/rki/coronawarnapp/greencertificate/ui/certificates/CertificatesViewModel.kt

```diff
@@ -1,11 +1,77 @@
 package de.rki.coronawarnapp.greencertificate.ui.certificates
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.asLiveData
 import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.greencertificate.ui.certificates.items.CertificatesItem
+import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import de.rki.coronawarnapp.vaccination.core.VaccinatedPerson
+import de.rki.coronawarnapp.vaccination.core.VaccinationSettings
+import de.rki.coronawarnapp.vaccination.core.repository.VaccinationRepository
+import de.rki.coronawarnapp.vaccination.ui.cards.BottomInfoVaccinationCard
+import de.rki.coronawarnapp.vaccination.ui.cards.CreateVaccinationCard
+import de.rki.coronawarnapp.vaccination.ui.cards.HeaderInfoVaccinationCard
+import de.rki.coronawarnapp.vaccination.ui.cards.ImmuneVaccinationCard
+import de.rki.coronawarnapp.vaccination.ui.cards.VaccinationCard
+import kotlinx.coroutines.flow.map
 
-class CertificatesViewModel @AssistedInject constructor() : CWAViewModel() {
+class CertificatesViewModel @AssistedInject constructor(
+    vaccinationRepository: VaccinationRepository,
+    private val vaccinationSettings: VaccinationSettings
+) : CWAViewModel() {
+
+    val events = SingleLiveEvent<CertificatesFragmentEvents>()
+
+    // TODO: cards should be adjusted in the following PR
+    val screenItems: LiveData<List<CertificatesItem>> =
```

Comment:
conside to move some code in functions for better readability of the code...

Reply:
I'll consider it in the following PR focused on cards.